### DISCUSSION TITLE
First Time Moose Install Compilation Error

### DISCUSSION THREAD

**Discussion by sem1042** [idaholab/moose#16242] [Q&A Getting Started] [answered]
I'm trying to install moose on MacOS and when i ran `make` I got the following error:

```
(moose) Sarahs-MacBook-Pro:test smoriarty$ make -j 4
Rebuilding symlinks in /Users/smoriarty/projects/moose/framework/build/header_symlinks
Rebuilding symlinks in /Users/smoriarty/projects/moose/test/build/header_symlinks
Creating Unity Directory /Users/smoriarty/projects/moose/framework/build/unity_src
Checking if header needs updating: /Users/smoriarty/projects/moose/framework/include/base/MooseRevision.h...
Copying default MOOSE configuration to: /Users/smoriarty/projects/moose/framework/include/base/MooseConfig.h...
Compiling C++ (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcrecpp.cc...
Compiling C++ (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_stringpiece.cc...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_compile.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_xclass.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_ucd.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_byte_order.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_jit_compile.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_get.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_exec.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_tables.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_string_utils.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_version.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_study.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_newline.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_refcount.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_maketables.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_config.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_fullinfo.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_dfa_exec.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_globals.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_ord2utf8.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_valid_utf8.c...
Compiling C (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/pcre/src/pcre_chartables.c...
Compiling C++ (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/gtest/gtest-all.cc...
Compiling C++ (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/hit/parse.cc...
Compiling C++ (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/hit/lex.cc...
Compiling C++ (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/hit/braceexpr.cc...
Building and linking /Users/smoriarty/projects/moose/framework/../python/pyhit/hit.so...
Compiling C++ (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/exodiff/exo_block.C...
Compiling C++ (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/exodiff/exodiff.C...
Compiling C++ (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/exodiff/add_to_log.C...
Compiling C++ (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/exodiff/ED_SystemInterface.C...
Compiling C++ (in opt mode) /Users/smoriarty/projects/moose/framework/contrib/exodiff/util.C...
ld: unsupported tapi file type '!tapi-tbd' in YAML file '/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib/libSystem.tbd' for architecture x86_64
clang-9: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [/Users/smoriarty/projects/moose/framework/../python/pyhit/hit.so] Error 1
make: *** Waiting for unfinished jobs....
```

Does anyone know what's going on?

**Comment by milljm** [comment]
~~Being discussed here: #16123~~

Edit: Mistaken for a different issue.

**Comment by dschwen** [comment]
Delete the apple xcode command line tools brute force first! 

```
rm -rf `xcode-select -p`
```

Install **version 11.5** of the apple command line tools from https://developer.apple.com/download/more/

**Reply by cticenhour** [reply]
FYI this issue will be fixed (and allow updates to CLT / XCode 12) as part of #16295. This is currently a work in progress - it will involve both conda recipe updates as well as a libmesh update.

**Reply by cticenhour** [reply]
Also this isn't exactly the answer if you have XCode installed over CLT...this would just remove the SDK directories within XCode, not XCode itself. In that case, XCode must be fully removed by deleting the app. But if you have only Command Line Tools, then this is fine.

**Reply by dschwen** [reply]
Yes, @cticenhour is right. I also edited my post: The reboot is not necessary.

**Reply by sem1042** [reply]
``rm: /Library/Developer/CommandLineTools/Library/Frameworks: Permission denied
rm: /Library/Developer/CommandLineTools/Library: Permission denied
rm: /Library/Developer/CommandLineTools: Permission denied
(moose) Sarahs-MacBook-Pro:~ smoriarty$ /Volumes/Command\ Line\ Developer\ Tools/Command\ Line\ Tools.pkg 
-bash: /Volumes/Command Line Developer Tools/Command Line Tools.pkg: Permission denied``

I keep getting permission denied notifications when i delete Xcode with brute force. Any ideas why this is happening? Thank you for the guidance, it's much appreciated. 

Also I'm trying to install the "Command Line Tools for Xcode 11.5" correct?

**Reply by cticenhour** [reply]
This is a permissions issue, since you are trying to modify system files. You need to modify the above command to the following: 

```sudo rm -rf `xcode-select -p` ```

MacOS will ask for your password in order to temporarily elevate your permissions, and then you should be able to remove it. 

"Command Line Tools for Xcode 11.5" is correct.

**Reply by sem1042** [reply]
last question, I'm getting the same permission denied error when I try to install the new Xcode. How do I override?

**Reply by cticenhour** [reply]
After opening the DMG file you get from Apple (it'll load a virtual drive on your desktop that you also open), then you can click on the PKG installer, where MacOS will ask (in graphical form, not on the Terminal) for your permission to install and a password to elevate your permissions. Can you elaborate on how you are installing and what you are seeing?

**Reply by sem1042** [reply]
I got it to work, thank you!

**Comment by cticenhour** [comment, marked as answer]
MOOSE and our conda package have now been updated to allow for newer versions of XCode and Command Line Tools. If this issue is experienced by others, the answer is now to update MOOSE (https://mooseframework.inl.gov/getting_started/installation/index.html#update) and the conda package to the newest version (activate your moose environment, then `conda update --all`). Please see the full announcement and instructions to upgrade MOOSE here: https://groups.google.com/g/moose-announce/c/t4ogAXgU26Y

If you have any further issues or questions, please feel free to reach out!
